### DISCUSSION TITLE
recognition of user-provided price ranges

### DISCUSSION THREAD

**Discussion by jimgrady** [deepgram/community#920] [General help] [answered]
We are having some trouble getting speech recognition to work correctly for spoken price ranges. We are trying to use some post-processing to deal with this, but having some difficulty. For example,

If a user says: "Between four hundred and seven hundred thousand dollars," the speech to text output is "between $400,700,000." It would be understandable if the output were "between 400 and 700,000" and that would be easy to deal with in post-processing. But it is doing a literal concatenation of the low and high end of the price range to form a nonsensical number.

Any tips on settings or other techniques to deal with this? Or do I just need to submit it as a feature request?

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Reply by jimgrady** [reply]
I actually realized the issue is caused by "smart formatting." In our case it's better to turn off smart formatting and let the LLM that is next in the processing chain figure out what the price ranges mean. I'll submit a feature request to improve smart formatting's handling of price ranges.

**Comment by jpvajda** [comment, marked as answer]
@jimgrady  I think in your case disabling Smart Formatting seems the like the way to go.  We've heard a lot of feedback on Smart Formatting not handling numbers correctly. 

You can try using the numerals feature, stand alone and see if this helps.: https://developers.deepgram.com/docs/numerals

As for a feature request, we have this improvement to smart formatting on our roadmap, but I don't know exactly when we'd deliver an improvement.
